improve docs for KeyboardAvoidingView

gwmccull opened the pull request:
## Motivation

The documentation for `KeyboardAvoidingView` was pretty thin. Tried to fill it out more and corrected a couple words.

## Test Plan

n/a

## Release Notes

[DOCS] [ENHANCEMENT] [KeyboardAvoidingView] - Improve the documentation for the props for KeyboardAvoidingView

### Update names of internal methods on KeyboardAvoidingView

* **Who does this affect**: Users that are manually calling the methods on KeyboardingAvoidingView.
* **How to migrate**: Add an underscore before the name of the method
* **Why make this breaking change**: These methods are not meant to be public. For example, the exposed `onLayout` function is not a prop that accepts a function like is typical of the rest of React Native but is the internal method that is called when the component's onLayout is triggered.
* **Severity (number of people affected x effort)**: Low

gwmccull commented:
The methods listed in the `KeyboardAvoidingView` docs are not meant to be publicly accessible.  Those are the functions that the component uses itself and are not meant to be called by the user.

I'd like to either hide them from appearing on the docs or add documentation that they should not be used.  What is the preferred approach?

janicduplessis commented:
@gwmccull Thanks for the PR! To hide the methods you can rename them to start with an underscore (_method).

janicduplessis commented on Libraries/Components/Keyboard/KeyboardAvoidingView.js:
```diff
@@ -43,15 +43,19 @@ type KeyboardChangeEvent = {
     ...ViewPropTypes,
+    /**
+     * Specify how the `KeyboardAvoidingView` will react to the presence of
+     * the keyboard. Will it adjust its height, position or bottom padding?
```
Can you phrase this not as a question. Something like "Specify if it will adjust ..."

gwmccull commented:
@janicduplessis thanks for having a look at my PR. I updated the text and renamed those functions.  Technically, this is a breaking change so I updated the PR with the breaking change text but it shouldn't affect too many people unless they're manually calling those functions

Let me know if anything else is needed
